### PR TITLE
Added/Updated release and contributing docs, release notes to rst file

### PR DESCRIPTION
**Pull request by Purg**
- Updated CONTRIBUTING doc file to be simpler and less of a confusing.
- Added documentation on release procedure, including steps on releasing to PYPI.
- Changed release notes files to rst extension for inclusion in sphinx documentation.

### CODE REVIEW

**Comment by Purg** [comment]
PTAL @kfieldho @mleotta Just want to check that that this seems sensible and I'm open to feedback.

File: docs/release_process.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - Rename the ``docs/release_notes/pending_release.md`` file to
+      ``docs/release_notes/v{VERSION}.md``, matching the value in the
+      ``VERSION`` file.
+
+Tag new version
+---------------
+Create a new git tag using the new version number (format:
+``v<MAJOR.<MINOR>.<PATCH>``) on the merge commit for the version update branch
+merger::
+    $ git tag -a -m "[Major|Minor|Patch] release v#.#.#
```
Comment (mleotta) [inline docs/release_process.rst]: missing closing quote here

File: docs/release_process.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+Major and Minor Releases
+^^^^^^^^^^^^^^^^^^^^^^^^
+1. Create a new branch off of the ``master`` branch named something like
+   ``release-[major,minor]-{NEW_VERSION}``.
+    - Increment patch value in ``VERSION`` file.
+    - Rename the ``docs/release_notes/pending_release.md`` file to
+      ``docs/release_notes/v{VERSION}.md``, matching the value in the
+      ``VERSION`` file.
+
```
Comment (mleotta) [inline docs/release_process.rst]: There should be step 2 (and maybe 3) here as well.  You need to merge the branch you created into master and then hard reset the release branch to point to master.

File: docs/release_process.rst (hunk 1 above)
Comment (Purg) [reply docs/release_process.rst]: Fixed.

File: docs/release_process.rst (hunk 2 above)
Comment (Purg) [reply docs/release_process.rst]: I did miss that! Thanks for the catch.

**Comment by Purg** [comment]
If there are no other objections to this, I think I might merge this today.

**Pull request merged**